client/telemetry/TelemetryEventClient.go

```diff
@@ -58,6 +58,7 @@ type TelemetryEventClientImpl struct {
 	InstalledAppRepository         repository2.InstalledAppRepository
 	userAttributesRepository       repository.UserAttributesRepository
 	cloudProviderIdentifierService cloudProviderIdentifier.ProviderIdentifierService
+	cloudProviderCache             string
```

Comment:
move this in a common config which can be used in future as well for similar cases